File: frontend/src/types/scryfall.ts
export type ScryfallSyncStatusValue = "NOT_STARTED" | "FETCHING" | "CURRENT" | "FAILED";

export interface ScryfallSyncStatus {
  status: ScryfallSyncStatusValue;
  bulk_data_type: string;
  source_updated_at: string | null;
  started_at: string | null;
  finished_at: string | null;
  card_count: number;
  error_message: string | null;
}

File: backend/app/schemas/scryfall.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ScryfallSyncStatusRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    status: str
    bulk_data_type: str
    source_updated_at: datetime | None
    started_at: datetime | None
    finished_at: datetime | None
    card_count: int
    error_message: str | None
